1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/domain/utils/fizzbuzz.ts

```diff
@@ -0,0 +1,25 @@
+import { ArrayHasInvalidNumbersException } from "../exceptions/array-has-invalid-numbers.exception";
+
+export function fizzbuzzReplacement (sequence: number[]): string {
+    return sequence.reduce<string>((result: string, number: number): string => {
+        const dividedByThree = number % 3 === 0;
+        const dividedByFive = number % 5 === 0;
+        const invalidNumber = number < 1 || number > 100;
+
+        if (invalidNumber) {
+            throw new ArrayHasInvalidNumbersException(`Invalid number`)
+        }
+
+        if (dividedByThree && dividedByFive) {
+            result += 'fizzbuzz'
```

Comment:
this is introducing mutability, do we need it for this solution ?

Reply:
Oh, I did it because I thought we would receive an array of numbers and then return a string concatenation. But fixed it in v2